contracts/transfer/circuits/src/execute/input.rs

```diff
@@ -0,0 +1,228 @@
+///
+/// Currently, Plonk is not a dependency of phoenix-core. This means the circuit
+/// construction of the note must be done here.
+///
+/// Ideally, there would be a `fn hash_inputs_witness(&self, composer)` implemented for
+/// `Note`.
+///
+/// Since the circuit will perform a pre-image check over the result of this function, the
+/// structure is safe
+///
+/// However, if `Note::hash_inputs` ever change, this circuit will be broken
+#[derive(Debug, Clone, Copy)]
+pub struct WitnessInput {
+    pub sk_r: Variable,
+    pub pk_r: Point,
+    pub note_hash: Variable,
+
+    pub note_type: Variable,
+    pub value_commitment: Point,
+    pub nonce: Variable,
+    pub r: Point,
+    pub pos: Variable,
+    pub cipher: [Variable; PoseidonCipher::cipher_size()],
+
+    pub value: Variable,
+    pub blinding_factor: Variable,
+
+    pub pk_r_prime: Point,
+    pub schnorr_message: Variable,
+    pub schnorr_u: Variable,
+    pub schnorr_r: Point,
+    pub schnorr_r_prime: Point,
+
+    // Public data
+    pub nullifier: BlsScalar,
+}
+
+impl WitnessInput {
+    pub fn to_hash_inputs(&self) -> [Variable; 12] {
+        [
+            self.note_type,
+            *self.value_commitment.x(),
+            *self.value_commitment.y(),
+            self.nonce,
+            *self.pk_r.x(),
+            *self.pk_r.y(),
+            *self.r.x(),
+            *self.r.y(),
+            self.pos,
+            self.cipher[0],
+            self.cipher[1],
+            self.cipher[2],
+        ]
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct CircuitInput<const DEPTH: usize> {
+    sk_r: JubJubScalar,
+    branch: PoseidonBranch<DEPTH>,
+    note: Note,
+    value: u64,
+    blinding_factor: JubJubScalar,
+    signature: Signature,
+    nullifier: BlsScalar,
+}
+
+impl<const DEPTH: usize> CircuitInput<DEPTH> {
+    pub fn new<R, L, A, S>(
+        rng: &mut R,
+        tree: &PoseidonTree<L, A, S, DEPTH>,
+        sk_r: JubJubScalar,
+        note: Note,
+        value: u64,
+        blinding_factor: JubJubScalar,
+        nullifier: BlsScalar,
+    ) -> Result<Self>
+    where
+        R: RngCore + CryptoRng,
+        L: PoseidonLeaf<S>,
+        A: PoseidonTreeAnnotation<L, S>,
+        S: Store,
+    {
+        let branch = tree
+            .branch(note.pos() as usize)
+            .map_err(|e| anyhow!("Poseidon tree error: {}", e))
+            .and_then(|branch| {
+                branch.ok_or(anyhow!(
+                    "The provided input note doesn't belong to the tree!"
+                ))
+            })?;
+
+        let message = ExecuteCircuit::<DEPTH>::sign_message();
+        let signature = SchnorrSecret::from(&sk_r).sign(rng, message);
+
+        Ok(Self {
+            sk_r,
+            branch,
+            note,
+            value,
+            blinding_factor,
+            signature,
+            nullifier,
+        })
+    }
+
+    pub fn note(&self) -> &Note {
+        &self.note
+    }
+
+    pub fn sk_r(&self) -> &JubJubScalar {
```

Comment:
Can these functions not be accessed as exports of dusk-pki or phoenix core to call fields? If I remember correctly they should be part of `Ownable` unless this structure has changed.

Reply:
This was only a helper method to behave as getter. But removed, it isn't necessary anywhere.